zebra-chain/src/transaction/serialize.rs

```diff
@@ -66,6 +66,65 @@ impl<P: ZkSnarkProof> ZcashDeserialize for Option<JoinSplitData<P>> {
     }
 }
 
+struct SigShieldedData<'a> {
+    sig: bool,
+    shielded_data: &'a Option<ShieldedData>,
+}
```

Comment:
`transaction::ShieldedData` includes `binding_sig`, for the Sapling data I'm not sure why this type is needed.

Reply:
It is needed for the order of the fields inside a transaction. In V4 transactions we have the spends and outputs for sapling then the joinsplits and then the binding sig. This is explained in https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/transaction/serialize.rs#L147-L153 and can be verified by checking the fields of the V4 transaction in the spec.

In the other hand in the V5 transactions the binding sig is intermediately after the sapling spends and outputs so it can be serialized right there as there is nothing in the middle.

Reply:
Please add a doc comment for this type, that explains why it's needed, and how it should be used.

Reply:
Actually, it looks like we can't use this type, because `ShieldedData` is different in V4 and V5 transactions.